src/AWS.Deploy.ServerMode.Client/ServerModeSession.cs

```diff
@@ -112,6 +129,30 @@ public ServerModeSession(int startPort = 10000, int endPort = 10100, string depl
             _deployToolPath = deployToolPath;
         }
 
+        public async Task AttachToRunnningSession(int port, CancellationToken cancellationToken)
```

Comment:
I'm uncomfortable having this method since it will be useless in the normal mode of server-mode running with encryption. It implies something more powerful then it really can do. Is the goal of this method just to make sure `_baseUrl` is set? If so maybe have `TryGetRestAPIClient` have a named parameter for `baseUrl`.

Reply:
I removed this method and moved some of the inner logic to the new `TryGetUnsecureRestAPIClient`. I did not want to add `baseUrl` as a named parameter as that is created in the `Start` method where we go over a bunch of available ports. I didn't want to make the user responsible for that. Plus adding that would have been a breaking change.